cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/WorkflowDataset.java

```diff
@@ -0,0 +1,289 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal.app.store;
+
+import co.cask.cdap.api.common.Bytes;
+import co.cask.cdap.api.dataset.lib.AbstractDataset;
+import co.cask.cdap.api.dataset.table.Row;
+import co.cask.cdap.api.dataset.table.Scan;
+import co.cask.cdap.api.dataset.table.Scanner;
+import co.cask.cdap.api.dataset.table.Table;
+import co.cask.cdap.common.utils.ImmutablePair;
+import co.cask.cdap.proto.Id;
+import co.cask.cdap.proto.ProgramType;
+import co.cask.cdap.proto.WorkflowStatistics;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Joiner;
+import com.google.common.primitives.Longs;
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import com.sun.istack.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Dataset for Completed Workflows and their associated actions
+ */
+public class WorkflowDataset extends AbstractDataset {
+
+  private static final Gson GSON = new Gson();
+  private static final byte[] RUNID = Bytes.toBytes("r");
+  private static final byte[] TIME_TAKEN = Bytes.toBytes("t");
+  private static final byte[] NODES = Bytes.toBytes("n");
+  private static final String DELIMITER = ":";
+  private static final Logger LOG = LoggerFactory.getLogger(WorkflowDataset.class);
+  private static final Type ACTION_RUNS_TYPE = new TypeToken<List<ActionRuns>>() { }.getType();
+
+  private final Table table;
+
+  WorkflowDataset(Table table) {
+    super("ignored", table);
```

Comment:
would having this instance name ``'ignored`` same as the MDS , affect in any way ? do you know ?